app/src/api.ts

```diff
@@ -0,0 +1,82 @@
+/** urls for this github repo */
+
+export const repo = "3b1b/captions";
+export const repoFull = `https://github.com/${repo}`;
+export const repoRaw = `https://raw.githubusercontent.com/${repo}`;
+export const website = "https://3blue1brown.com";
+export const websiteRaw =
+  "https://raw.githubusercontent.com/3b1b/3Blue1Brown.com";
+export const cloudFuncBase =
+  "https://us-central1-mystic-fountain-606.cloudfunctions.net/test-function";
+
+/** generic request */
+export async function request<T>(
+  url: string,
+  options: RequestInit = {},
+  format: "json" | "text" = "json",
+) {
+  try {
+    const response = await fetch(url, options);
+    if (!response.ok) throw Error(String(response.status));
+    return (await (await fetch(url))[format]()) as T;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+/** check for presence of branch */
+export async function branchExists(name: string) {
+  const url = `https://api.github.com/repos/${repo}/branches/${name}`;
+  /** seemingly no credentials needed for just reading/checking */
```

Comment:
No credentials needed, but I think with heavy use GitHub might rate limit the user's IP. Might want to make an cloud function endpoint for checking this, at some point.